frontends/web/src/components/password.jsx

```diff
@@ -309,3 +306,13 @@ function MatchesPattern({ regex, value = '', text }) {
         <p style="color: var(--color-error);">{text}</p>
     );
 }
+
+const excludeKeys = /^(Shift|Alt|Backspace|CapsLock|Tab)$/i;
+
+function hasCaps({ key }) {
+    if (key.length > 1 || key.toUpperCase() === key.toLowerCase() || excludeKeys.test(key)) {
+        return null;
```

Comment:
Return `false` instead of `null`? Then the caller can just `setState()` it without a null check.

Reply:
Seems like it's a tri-state return value?
Maybe `hasCaps` func needs a little doc comment, describing what it returns in which case.